src/example/Example_AdaptiveController.cpp

```diff
@@ -128,9 +129,57 @@ std::tuple<VectorXd,double> closest_invariant_error(const DQ& x, const DQ& xd, c
         }
         return {vec8(ex),invariant};
     }
+    }
     throw std::runtime_error("Not supposed to be reachable");
 }
 
+
+/*
+    get the error between the desired target primitive (xd, in world frame) and the primitive (primitive_, in end-effector frame)
+    x:  pose of the robot, wrt world frame
+    xd: desired objective pose of the objective
+    control_objective_: type of the objective
+    primitive_: pose of the objective, wrt end-effector
+*/
+std::tuple<VectorXd,double> closest_invariant_primitive_error(const DQ& x, const DQ& xd, const ControlObjective& control_objective_, const DQ& primitive_)
```

Comment:
@LiuzhengSJ Please don't change the name of the function, and just add the primitive as having a default value to consider cases in which it is not needed.

for instance
`const DQ& primitive_ = DQ(1)`

Reply:
Thanks, @LiuzhengSJ. I understand how this works but that's not the modification I'm requesting. I'll explain in the main body of the issue.